lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -46,7 +46,6 @@ def matches?(given_proc, negative_expectation=false, &block)
           @eval_block = false
           @eval_block_passed = false
 
-          warn_about_bare_error if warning_about_bare_error && !negative_expectation
           return false unless Proc === given_proc
```

Comment:
Yeah I was surprised that the build did'nt fail after moving that line of code, but I thought that's the cleanest/simplest way of capturing what the actual_error was.

Maybe I could move that line back and then update the warning message when `@actual_error` is not `nil`?